### PR TITLE
Removed packages that already exist, updated README

### PR DESCRIPTION
**Pull request by leahnp**
Removed packages from build/imagerun.sh that were already added by the Dockerfile. 

Added more information to README.

Fixes https://github.com/samsung-cnct/k2-tools/issues/66

### CODE REVIEW

File: build/imagerun.sh [hunk 1]
```diff
@@ -5,22 +5,12 @@ set -o pipefail
 # Adding baseline alpine packages
 apk update 
 apk add --no-cache  \
-    bash \
-    ca-certificates \
     g++ \
     gcc \
-    libffi-dev \
```
Comment (coffeepac) [inline build/imagerun.sh]: can you add all of the apk packages from this list to the Dockerfile?  Or move them all here.  Either way is fine but having it in two locations is unnecessary.

And either way, ensure the two `rm -rf` lines at the bottom of this script are immediately following the `apk del` line.

File: build/imagerun.sh [hunk 2]
```diff
@@ -5,22 +5,12 @@ set -o pipefail
...
-    py-cryptography \
-    py-pip \
-    python \
-    python-dev \
-    unzip \
-    zip
+    python-dev 
 
 # Python / ansible addon work
 # update pip
```
Comment (coffeepac) [inline build/imagerun.sh]: if the only thing that's in this script is pip, please move the pip commands to the Dockerfile.  also, please add `awscli` to requirements.txt

Comment (leahnp) [inline build/imagerun.sh]: So completely remove imagerun.sh?

Comment (coffeepac) [inline build/imagerun.sh]: if you decide to move all of the apk activity into the Dockerfile then yes, remove imagerun.sh

**Comment by leahnp** [comment]
@coffeepac  made another pass at it - do I need to delete all the packages if k2-tools is only used for up/down/update? Is there an advantage to removing everything, even though it will just be killed after the process is done. It feels unnecessary, unless I am missing something.

**Comment by coffeepac** [comment]
@leahnp the package delete is important because it shrinks the size of the image.  the smaller the image size the faster it downloads to a user's computer.  its purely a user-experience thing.

**Comment by leahnp** [comment]
But it all ultimately is downloaded (meaning all the packages we install at some point during the life of the container  - they must be downloaded to perform other tasks) but then we delete them (which I see as extra work, since we already downloaded them) then we kill the container pretty quickly once the k2 task is done.

I have no problem adding the delete line back in, I just feel like I am missing some information because it logically doesn't make sense to me.

**Comment by coffeepac** [comment]
the Dockerfile describes how to build the k2-tools image.  When we build that image we download all packages, install them, clean up the image and then we send the fully built/cleaned image up to quay for hosting.  

When we create a container based on the k2-tools image we do not execute the Dockerfile, that work is already done.  So when we create the container we want the image to be as small as possible.  Does that make sense?

**Comment by leahnp** [comment]
@coffeepac thanks for your help with this - build related packages are now removed after use.

**Comment by coffeepac** [comment]
Code looks fine but CI is failing.  Looks like openssl isn't being installed?

**Comment by leahnp** [comment]
@coffeepac I am checking out the CI - what makes you think openssl is not being installed?

**Comment by coffeepac** [comment]
the error message is:  

> {"changed": false, "cmd": "openssl genrsa -out /root/kraken/cluster/aws//pr-90-3/certs/ca-key.pem 2048", "failed": true, "msg": "[Errno 2] No such file or directory", "rc": 2}

Which means either the executable, openssl, doesn't exist or the output directory doesn't exist.  I'm assuming its openssl because there is an ansible task right before this error that creates the directory.

**Comment by leahnp** [comment]
Openssl is added on line 45 of the Dockerfile - and I logged into this version of k2-tools and see this output: ```/ # find / -name openssl
/usr/lib/python2.7/site-packages/cryptography/hazmat/bindings/openssl
/usr/lib/python2.7/site-packages/cryptography/hazmat/backends/openssl``` is that not the correct openssl?

**Comment by coffeepac** [comment]
looks fine!  can you execute it?

**Comment by leahnp** [comment]
Seems that was a red herring - it appears that was just a binding for a different python package for openssl. So back to your original theory that openssl is not being installed...even though its in the list of stuff being installed in the Dockerfile. Investigating more.

**Comment by leahnp** [comment]
It seemed that when alpine-sdk was deleted, it was also deleting openssl. Let's see if this passes CI

**Pull request merged**